cylc/flow/suite_files.py

```diff
@@ -306,58 +352,61 @@ def get_contact_file(reg):
+    """Locate/load Curve private-key/ ...etc.
 
     Return file name, or content of file if content=True is set.
     Files are searched from these locations in order:
 
-    1/ For running task jobs, service directory under:
+    1/  a/ Server Curve ZMQ keys located in:
+            suite service directory/server_keys/private
+            suite service directory/server_keys/public
+
+        b/ Client Curve ZMQ keys located in:
+            suite service directory/client_keys/private
+            suite service directory/client_keys/public
+
+    2/ For running task jobs, service directory under:
        a/ $CYLC_SUITE_RUN_DIR for remote jobs.
        b/ $CYLC_SUITE_RUN_DIR_ON_SUITE_HOST for local jobs or remote jobs
           with SSH messaging.
 
-    2/ For suite on local user@host. The suite service directory.
+    3/ For suite on local user@host. The suite service directory.
 
-    3/ Location under $HOME/.cylc/ for remote suite control from accounts
+    4/ Location under $HOME/.cylc/ for remote suite control from accounts
        that do not actually need the suite definition directory to be
        installed:
        $HOME/.cylc/auth/SUITE_OWNER@SUITE_HOST/SUITE_NAME/
 
-    4/ For remote suites, try locating the file from the suite service
+    5/ For remote suites, try locating the file from the suite service
        directory on remote owner@host via SSH. If content=False, the value
        of the located file will be dumped under:
        $HOME/.cylc/auth/SUITE_OWNER@SUITE_HOST/SUITE_NAME/
 
     """
     if item not in [
-            SuiteFiles.Service.PASSPHRASE, SuiteFiles.Service.CONTACT,
-            SuiteFiles.Service.CONTACT2,
-            UserFiles.Auth.SERVER_PRIVATE_KEY_CERTIFICATE,
-            UserFiles.Auth.SERVER_PUBLIC_KEY_CERTIFICATE]:
-        raise ValueError("%s: item not recognised" % item)
-
-    # For a UserFiles.Auth.SERVER_..._KEY_CERTIFICATE, only need to check Case
-    # '3/' (always ignore content i.e. content=False), so check these first:
-    if item in (UserFiles.Auth.SERVER_PRIVATE_KEY_CERTIFICATE,
-                UserFiles.Auth.SERVER_PUBLIC_KEY_CERTIFICATE):
-        auth_path = os.path.join(
-            get_suite_srv_dir(reg), UserFiles.Auth.DIRNAME)
-        public_key_dir, private_key_dir = return_key_locations(auth_path)
-        if item == UserFiles.Auth.SERVER_PRIVATE_KEY_CERTIFICATE:
-            path = private_key_dir
-        else:
-            path = public_key_dir
-        value = _locate_item(item, path)
-        if value:
-            return value
+            SuiteFiles.Service.CONTACT,
+            SuiteFiles.Service.CONTACT2] and not isinstance(item, KeyInfo):
+        raise ValueError(f"{item}: item not recognised")
+
+        # 1 (a)
+    if isinstance(item, KeyInfo):
+
+        item_location = _locate_item(item.file_name, item.key_path)
+
+        # Temporary hack until we can separate key file 'get' into own function
```

Comment:
Add a TODO for it?

Reply:
Yes, will do. I think the get_auth_item method could do with a rework in the near future. Perhaps with the passphrase removal ticket: authentication: wash up #3449.